tests/general/base.py

```diff
@@ -0,0 +1,200 @@
+from test_plus import APITestCase
+from django.core.cache import cache
+
+from tests.profile.factories import (
+    ProfileFactory, ProfileHighlightFactory,
+    ProfileKeyMetricsFactory, IndicatorCategoryFactory,
+    IndicatorSubcategoryFactory
+)
+from tests.points.factories import (
+    ProfileCategoryFactory, ThemeFactory, CategoryFactory, LocationFactory
+)
+from tests.datasets.factories import (
+    GeographyFactory, GeographyHierarchyFactory,
+    VersionFactory, DatasetFactory, GroupFactory,
+    DatasetDataFactory, IndicatorFactory
+)
+from tests.boundaries.factories import GeographyBoundaryFactory
+from wazimap_ng.datasets.models import Group, Geography
+from wazimap_ng.datasets.tasks.indicator_data_extraction import (
+    indicator_data_extraction
+)
+
+
+
+class ConsolidatedProfileViewBase(APITestCase):
+
+    def setUp(self):
+        self.version = VersionFactory()
+        self.root = Geography.add_root(code="ZA", level="country")
+        self.geo1 = self.root.add_child(code="WC", level="province")
+        self.geo2 = self.root.add_child(code="EC", level="province")
+        self.dc1 = self.geo1.add_child(code="DC1", level="muni")
+        self.create_boundary(self.root, self.version)
+        self.create_boundary(self.geo1, self.version)
+        self.create_boundary(self.geo2, self.version)
+
+        self.hierarchy = self.create_hierarchy(
+            self.root, self.version
+        )
+        self.profile = ProfileFactory(geography_hierarchy=self.hierarchy)
+        self.category = CategoryFactory(profile=self.profile)
+        self.location = LocationFactory(category=self.category)
```

Comment:
does anything use this? It looks out of place. I know it's just moved, but if it passes without it, we can drop it.

Reply:
Yes we are need locations for themes-count api. Theme is excluded if there is no location in it.